sdk/daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Speedy.scala

```diff
@@ -1315,6 +1336,19 @@ private[lf] object Speedy {
             ),
           svalue => Control.Value(svalue),
         )
+
+    final def updateGasBudget(cost: CostModel => CostModel.Cost): Unit =
+      if (hasGasBudget) {
+        val consumed = cost(costModel)
+        if (consumed > 0) {
```

Comment:
Should we do anything if `consumed` is negative here? Worried that a programmer error might break things in some weird way?